source/Calamari/Kubernetes/ResourceStatus/ResourceStatusReportWrapper.cs

```diff
@@ -59,120 +49,24 @@ public bool IsEnabled(ScriptSyntax syntax)
             ICommandLineRunner commandLineRunner,
             Dictionary<string, string> environmentVars)
         {
-            var customKubectlExecutable = variables.Get(SpecialVariables.CustomKubectlExecutable);
-            var deploymentTimeoutSeconds = variables.GetInt32(SpecialVariables.DeploymentTimeout) ?? 0;
-            var stabilizationTimeoutSeconds = variables.GetInt32(SpecialVariables.StabilizationTimeout) ?? 0;
-            var defaultNamespace = variables.Get(SpecialVariables.Namespace, "default");
             var workingDirectory = Path.GetDirectoryName(script.File);
-            
+
             var result = NextWrapper.ExecuteScript(script, scriptSyntax, commandLineRunner, environmentVars);
             if (result.ExitCode != 0)
             {
                 return result;
             }
 
-            var manifests = ReadManifestFiles().ToList();
-            var definedResources = KubernetesYaml.GetDefinedResources(manifests, defaultNamespace).ToList();
-            
-            var secret = GetSecret(defaultNamespace);
-            if (secret != null)
+            try
             {
-                definedResources.Add(secret);
+                statusReportExecutor.ReportStatus(workingDirectory, commandLineRunner, environmentVars);
```

Comment:
All of the removed stuff in this method has been moved to `ResourceStatusReportExecutor` which is a new class.